src/main/java/org/nuxeo/ecm/core/storage/marklogic/MarkLogicRepository.java

```diff
@@ -672,13 +682,21 @@ private State findOne(String ctsQuery) {
             logQuery(query);
         }
         // Run query
-        try (Session session = xccContentSource.newSession()) {
-            AdhocQuery request = session.newAdhocQuery(query);
-            // ResultSequence will be closed by Session close
+        try {
+            // Don't auto close session it is the stream responsibility
+            Session session = xccContentSource.newSession();
```

Comment:
Indeed 👍 , nevertheless I will rework the stream initialisation to fully benefit try-with-resource feature as it's just one method which could throw an exception. Other calls may not throw exception.